cpu/stm32f4/periph/dac.c

```diff
@@ -22,145 +24,40 @@
 #include "periph/dac.h"
 #include "periph_conf.h"
 
-/* guard in case that no DAC device is defined */
-#if DAC_NUMOF
```

Comment:
nope! This is one of the ideas of the re-doing of periph interfaces to get rid of this anti-pattern. At least at some point the approach of #3420 will remove the need of these guards once and for all